### PR TITLE
Login UI

### LINKED ISSUE #11: GUI for Log In
- Includes booking step 2 (if user wants to log in)

Design in figma:

https://www.figma.com/file/bRPboJBlptgd24c9WWHxEH/UI%2FUX---Gruppuppgift-2?t=ASDDbNx3p1iKnX7A-0

### PR DESCRIPTION
**Pull request by jakobpalmborg**
# UI for login page

- Form for login
- SCSS for login page
- Handle input from form
- Link to sign up page
- Classes for primary-btn and form-input-field in global.scss

image

### CODE REVIEW

**Review by OlliverLar — APPROVED** [review]
Ser bra ut Jakob

File: src/app/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <section className="sec-cont login-container">
+      <div>
+        <h1>Inloggning</h1>
+        <form action="">
+          <label htmlFor="email">Email</label>
+          <input
+            onChange={(e) => {
+              e.preventDefault();
```
Comment (rfabricioflores) [inline src/app/login/page.tsx]: The prevent default doesn't go here, it has to be in the form using the onSubmit event.

**Tips**
This is another way to update the useState with less code:
```tsx
const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setUserData({
      ...userData,
      [e.target.id]: e.currentTarget.value,
    });
};
  
<input
    onChange={handleChange}
    className="form-input-field"
    type="text"
    id="email"
/>

<input
    onChange={handleChange}
    className="form-input-field"
    type="password"
    id="password"
/>
```

File: src/app/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+              });
+            }}
+            className="form-input-field"
+            type="text"
+            id="email"
+          />
+          <label htmlFor="password">Password</label>
+          <input
+            onChange={(e) => {
+              e.preventDefault();
```
Comment (rfabricioflores) [inline src/app/login/page.tsx:33]: The prevent default doesn't go here, it has to be in the form using the onSubmit event.

File: src/app/login/page.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const [userData, setUserData] = useState({
+    email: '',
+    password: '',
+  });
+
+  return (
+    <section className="sec-cont login-container">
+      <div>
+        <h1>Inloggning</h1>
+        <form action="">
```
Comment (rfabricioflores) [inline src/app/login/page.tsx:17]: Prevent default goes here.
Use this:
```tsx
<form
     action=""
     onSubmit={(e) => {
       e.preventDefault();
       console.log(userData);
     }}
></form>
```

**Review by rfabricioflores — APPROVED** [review]

**Comment by rfabricioflores** [comment]
The UI looks great, but there are a few small changes I recommend you to make before merging ✅

File: src/app/login/page.tsx (hunk 1 above)
Comment (jakobpalmborg) [reply src/app/login/page.tsx]: I took your shorter syntax for setUserData but I like to have the event inside jsx so typescript can use type inference:
image

File: src/app/login/page.tsx (hunk 2 above)
Comment (jakobpalmborg) [reply src/app/login/page.tsx:33]: yes, of course!

File: src/app/login/page.tsx (hunk 3 above)
Comment (jakobpalmborg) [reply src/app/login/page.tsx:17]: I think we can write the submit when we write the logic for submitting the form, in another PR.

**Pull request merged**